samples/nrf9160/modem_shell/README.rst

```diff
@@ -0,0 +1,380 @@
+
+Examples
+--------
+
+Open and connect to an IP address and port (IPv4 TCP socket)::
+
+   sock connect -a 111.222.111.222 -p 20000
+
+Open and connect to hostname and port (IPv4 TCP socket)::
+
+   sock connect -a google.com -p 20000
+
+Open and connect IPv6 TCP socket and bind to a port::
+
+   sock connect -a 1a2b:1a2b:1a2b:1a2b::1 -p 20000 -f inet6 -t stream -b 40000
+
+Open IPv6 UDP socket::
+
+   sock connect -a 1a2b:1a2b:1a2b:1a2b::1 -p 20000 -f inet6 -t dgram
+
+Open raw socket::
+
+   sock connect -f packet -t raw
+
+Open a socket to non-default PDP context::
+
+   ltelc connect -a nondefault.context.com
+   sock connect -f packet -t raw -I 1
+
+Send string through socket::
+
+   sock send -i 0 -d testing
+
+Send 100kB of data and show throughput statistics::
+
+   sock send -i 0 -l 100000
+
+Send data periodically with 10s interval::
+
+   sock send -i 0 -e 10 -d test_periodic
+
+Calculate receive throughput::
+
+   <do whatever is needed to make device receive data after some time>
+   sock recv -i 0 -r -l 1000000
+   sock recv -i 0
+   sock recv -i 0
+
+Close socket::
+
+   sock close -i 0
+
+Use RAI settings::
+
+   ltelc funmode -4
+   sock rai --rai_enable
+   ltelc funmode -1
+   sock connect -a 111.222.111.222 -p 20000
+   sock rai -i 0 --rai_last
+   sock send -i 0 -d testing
+
+List open sockets::
+
+   sock list
+
+SMS tool
+========
+
+MoSh command: "sms"
+
+SMS tool can be used for sending and receiving SMS messages.
+
+Examples
+--------
+
+Register SMS service so messages can be received if SIM subscription supports it:
+
+   sms reg
+
+Send SMS message (registration is done automatically if not already done)::
+
+   sms send -n +987654321 -m testing
+
+GNSS
+====
+
+MoSh command: "gnss"
+
+GNSS provides commands for searching the location of the device.
+
+Examples
+--------
+
+Start GNSS tracking and then stop it::
+
+   gnss start
+   gnss stop
+
+Disable LTE, enable all GNSS output and start continuous tracking with power saving enabled::
+
+   ltelc funmode --lteoff
+   gnss output 2 1 1
+   gnss mode cont
+   gnss config powersave perf
+   gnss start
+
+Enable LTE PSM, only NMEA output, automatic A-GPS data fetching and start periodic fixes with 5 minute interval and 120 second timeout::
+
+   ltelc psm -e
+   gnss output 0 1 0
+   gnss agps automatic enable
+   gnss mode periodic 300 120
+   gnss start
+
+Enable Low Noise Amplifier (LNA) for use with the on-board GPS antenna::
+
+   gnss lna enable
+
+FOTA
+====
+
+MoSh command: "fota"
+
+FOTA can be used to perform software updates over-the-air. However, you need to know what updates are available in NRF servers in order to use this feature completely.
+
+PPP
+===
+
+MoSh command: "ppp"
+
+PPP (Point-to-Point Protocol) can be used to enable dial-up access to the Internet.
+MoSh command is simple as shown in below example but you need to have normal dial up setup in your PC in order to use development kit's PPP interface.
```

Comment:
What do we do with PPP documentation? MoSh shell command is simple “ppp up” but the setup is more complicated and not something normal.

Reply:
By the "setup" do you mean host-side configuration? IMO we should provide a guide for this as well, although I'm not sure how complex that is. Perhaps there's some tutorial on the internet already we could link to? It should be pretty generic setup I guess?

Reply:
Yes, I mean host-side configuration. You've got a link to the setup description in your mailbox.

Reply:
I've added this note as agreed:
```
.. note::
   Dial-up connection is non functional when using Segger virtual UART ports.
   PPP has been used successfully with FTDI UART port though.
   Refer to Section 2.1.2 in nRF9160 Hardware Verification Guidelines (nWP-034):
   https://infocenter.nordicsemi.com/pdf/nwp_034.pdf
```